apps/docs/src/guide/provider/rpc-consistency.md

```diff
@@ -0,0 +1,11 @@
+# RPC Consistency
+
+A common issue with querying distributed networks is ensuring each node is consistent. At any moment each node could be at a different block height, therefore querying the state of the blockchain may yield different results from node to node.
+
+To defend against this, the SDK appends the block height to block-sensitive requests. The node will then ensure it has met the block height criteria before it processes the request.
+
+If the block height criteria has not been met by the node, the request will either be awaited on the node's side (if the node supports this), or it'll be retried from the SDK side. The SDK will attempt to retry the request until it hits a maximum amount of retries, and then will throw an error.
```

Comment:
Should we include information here about the values used by the TS SDK for retry attempts?

```ts
maxRetries: 5,
baseDelay: 500,
```

Also, it would be helpful to mention that `Provider.clearChainAndNodeCaches` resets the `currentBlockHeightCache` for all provider instances associated with the given URL or for all URLs when a specific URL is not informed.

On a related note, we should also clarify that all Provider instances created with the same URL share the same `currentBlockHeightCache`. Including code snippets with multiple provider instances could help make this behavior easier to understand.

Reply:
Duplicated